app/ui/legacy/src/main/java/com/fsck/k9/account/AccountRemoverWorker.kt

```diff
@@ -0,0 +1,48 @@
+package com.fsck.k9.account
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.Data
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+/**
+ * An [AccountRemoverWorker] to remove an account in the background.
+ */
+class AccountRemoverWorker(
+    context: Context,
+    workerParams: WorkerParameters,
+) : CoroutineWorker(context, workerParams), KoinComponent {
```

Comment:
We're not calling a suspending function in `doWork()`. So using a regular `Worker` will do.